lib/Parser/Parse.cpp

```diff
@@ -5814,52 +5796,53 @@ void Parser::ParseTopLevelDeferredFunc(ParseNodeFnc * pnodeFnc, ParseNodeFnc * p
 
     m_ppnodeVar = &pnodeFnc->pnodeVars;
 
-    if (pnodeFncParent != nullptr
-        && m_currDeferredStub != nullptr
-        // We don't create stubs for function bodies in parameter scope.
-        && pnodeFnc->pnodeScopes->blockType != PnodeBlockType::Parameter)
+    // Don't try and skip scanning nested deferred lambdas which have only a single expression in the body.
+    // Their more-complicated text extents won't match the deferred-stub and the single expression should be fast to scan, anyway.
+    if (fLambda && !*pNeedScanRCurly)
+    {
+        ParseExpressionLambdaBody<false>(pnodeFnc);
+    }
+    else if (pnodeFncParent != nullptr && m_currDeferredStub != nullptr)
     {
         // We've already parsed this function body for syntax errors on the initial parse of the script.
         // We have information that allows us to skip it, so do so.
 
+        Assert(pnodeFncParent->nestedCount != 0);
         DeferredFunctionStub *stub = m_currDeferredStub + (pnodeFncParent->nestedCount - 1);
-        Assert(pnodeFnc->ichMin == stub->ichMin);
+
+        Assert(pnodeFnc->ichMin == stub->ichMin
+            || (stub->fncFlags & kFunctionIsAsync) == kFunctionIsAsync
+            || ((stub->fncFlags & kFunctionIsGenerator) == kFunctionIsGenerator && (stub->fncFlags & kFunctionIsMethod) == kFunctionIsMethod));
+
         if (stub->fncFlags & kFunctionCallsEval)
         {
             this->MarkEvalCaller();
         }
-        if (stub->fncFlags & kFunctionChildCallsEval)
```

Comment:
Why do we no longer need these?

Reply:
Yes, I just copied all the flags from the stub instead of picking and choosing them.

Reply:
Sounds good. Thanks.